components/Elevation/src/UIColor+MaterialElevation.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface UIColor (MaterialElevation)
+
+/// Returns a color that takes the specified elevation value into account.
+/// @param elevation The elevation to use when resolving the color.
+- (UIColor *)resolvedColorWithElevation:(CGFloat)elevation;
+
+/// Returns a color that takes the specified elevation value and traits into account.
+/// @param traitCollection The traits to use when resolving the color.
+/// @param elevation The elevation to use when resolving the color.
+- (UIColor *)resolvedColorWithTraitCollection:(UITraitCollection *)traitCollection
```

Comment:
Can we match the Typography method ([`mdc_scaledFontForTraitEnvironment:`](https://github.com/material-components/material-components-ios/blob/ec4b19d/components/Typography/src/UIFont%2BMaterialScalable.h#L64)) and pass a `UITraitEnvironment` here instead? Is there a specific reason why we would pass a `UITraitCollection` rather than an environment?

Reply:
This is on par with the new iOS 13 APIs for resolving colors based on traitCollection, with the addition of elevation in our case. Please see: https://developer.apple.com/documentation/uikit/uicolor/3238042-resolvedcolorwithtraitcollection?language=objc

Reply:
That's a good reason. Thanks for explaining. :)